WireDomain/Sources/WireDomain/Synchronization/OneOnOneResolver.swift

```diff
@@ -84,29 +87,33 @@ public struct OneOnOneResolver: OneOnOneResolverProtocol {
         let commonProtocol = await getCommonProtocol(between: selfUser, and: user)
 
         if mlsProvider.isMLSEnabled, commonProtocol == .mls {
-            try await resolveMLSConversation(
+            let groupId = try await resolveMLSConversation(
                 for: user
             )
+            action = .migratedToMLSGroup(identifier: groupId)
         }
 
         if mlsProvider.isMLSEnabled, commonProtocol == nil {
             await resolveNoCommonProtocolConversation(
                 between: selfUser,
                 and: user
             )
+            action = .archivedAsReadOnly
         }
 
         if commonProtocol == .proteus {
             await resolveProteusConversation(
                 for: user
             )
         }
+        return action
     }
 
-    private func resolveMLSConversation(for user: ZMUser) async throws {
+    @discardableResult
+    private func resolveMLSConversation(for user: ZMUser) async throws -> MLSGroupID {
         WireLogger.conversation.debug("Should resolve to mls 1-1 conversation")
 
-        let userID = await context.perform {
+        let userID = await context.perform { [user] in
```

Comment:
Capturing a managed object won't solve the `@Sendable` warning. I think the only way to silence it is to pass the object ID and fetch it in the context within the perform block.